cmd/countylookup/main.go

```diff
@@ -1,30 +1,70 @@
 package main
 
 import (
+	"encoding/csv"
+	"flag"
 	"fmt"
 	"os"
 
 	"github.com/corbaltcode/usps/internal/counties"
 )
 
 func main() {
-	if len(os.Args) < 2 {
-		fmt.Fprintf(os.Stderr, "Error: Missing tar file name\nUsage: %s <tar_file_name>\n", os.Args[0])
+	tarName := flag.String("tar", "", "Name of the tar file")
+	zipCode := flag.String("zip", "", "Single zip to look up (optional)")
+	flag.Parse()
+
+	if *tarName == "" {
+		fmt.Fprintf(os.Stderr, "Error: Missing tar file name\nUsage: %s -tar <tar_file_name> [-zip <optional_single_zip_code>]\n", os.Args[0])
 		os.Exit(1)
 	}
 
-	tarName := os.Args[1]
-
 	zipPassword := mustGetenv("ZIP_PASSWORD")
 
-	zipToCounty, err := counties.CollectZip4Details(tarName, zipPassword)
+	zipToCounty, err := counties.CollectZip4Details(*tarName, zipPassword)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to collect details: %v\n", err)
 		os.Exit(1)
 	}
 
+	// Early return for the user-specified, single zip case.
+	if *zipCode != "" {
+		printCountiesForZIP(zipToCounty, *zipCode)
+		return
+	}
+
+	err = exportToCSV(zipToCounty)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to export to CSV: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func exportToCSV(zipToCounty map[string][]string) error {
+	writer := os.Stdout
+	csvWriter := csv.NewWriter(writer)
```

Comment:
You could save a line of code without sacrificing much by "inlining" `writer` here: `csvWriter := csv.NewWriter(os.Stdout)`.